recipes/service.rb

```diff
@@ -68,8 +68,9 @@
   mode '644'
   variables(
     splunkdir: splunk_dir,
+    splunkuser: splunk_runas_user,
     splunkcmd: splunk_cmd,
-    runasroot: false,
+    runasroot: node['splunk']['server']['runasroot'],
```

Comment:
simply passing the attribute here may get false positives, because this merely equates to "true if the attribute exists" . The attribute could exist and have the value of `false` and this would still be "true" in ruby.

I'd prefer this:
```
runasroot: node['splunk']['server']['runasroot'] == 'root',
```

It might be time to add a helper method called `run_as_root` that does this comparison and returns a true/false value.